.github/workflows/tests.yml

```diff
@@ -26,7 +26,7 @@ jobs:
       matrix:
         python-version: ["3.6", "3.12"]
         # NOTE: align with versions in noxfile.py:
-        cocotb-version: ["1.6.0", "1.9.0"]
+        cocotb-version: ["1.6.0", "1.9.0", "github-828d127e"]
```

Comment:
Can we just use `"master"` here?

Reply:
https://github.com/cocotb/cocotb/commit/bfd811bbeba4c7dcc045beeb5c78f8e4901c8364 breaks something, thus we cannot use master (at least as of 2 weeks ago).

My plan was to get this PR through review and then, once the code is verified to be correct, raise a bug on cocotb repository.

Reply:
Other than that, I would actually recommend against using simple `master`, because this makes tests unstable. A future commit may break cocotb-bus CI. Then whoever makes a PR will be forced to fix this issue as well. Better to pin the dependencies and fix issues at our own speed.

Reply:
Alright. The break is that setimmediate works as intended now. All setimmediate in cocotb-bus should be updated to regular value setting since that's what it was doing before.

Reply:
Fixed now and also upgraded cocotb to latest master.